tests/console/zdup_prepare_leap_sle.pm

```diff
@@ -0,0 +1,103 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: prepare migration from Leap to SLE
+#   Documentation for 15:
+#   https://www.suse.com/de-de/documentation/sles-15/book_sle_upgrade/data/sec_upgrade-online_opensuse_to_sle.html
+#   12 is not supported and inoffical
+# Maintainer: lemon.li <leli@suse.com>
+
+use base "installbasetest";
+use registration;
+use testapi;
+use utils qw(zypper_call);
+use strict;
+use warnings;
+
+#   Function: judge base system is which version of leap
+#   parameters:
+#   $version : expected leap version
+#   return : 1 if ture
+sub is_leap_base {
+    my $version = shift;
+    return (get_var('HDDVERSION', '') =~ /opensuse-$version/);
+}
+
+sub run {
+    my $reg_code   = get_required_var('SCC_REGCODE');
+    my $scc_url    = get_required_var('SCC_URL');
+    my $scc_addons = get_var('SCC_ADDONS', '');
+
+    # so if we are migrating from Leap to SLE we just make SUSEConnect believe
+    # we are on SLE
+    my $hddversion = get_var('HDDVERSION', '');
+    my $version    = get_var('VERSION');
+    $version =~ s/-SP/./;    # 12-SP3 -> 12.3
```

Comment:
There is `scc_version` method in registration library, I would recommend to use that one.

Reply:
Yes, replace with scc_version. Thanks.